dash/testing/application_runners.py

```diff
@@ -261,13 +267,18 @@ def start(self, app, start_timeout=2, cwd=None):
                 cwd = os.path.dirname(app)
                 logger.info("RRunner inferred cwd from app path: %s", cwd)
         else:
-            path = (
-                "/tmp/app_{}.R".format(uuid.uuid4().hex)
-                if not self.is_windows
-                else os.path.join(
-                    (os.getenv("TEMP"), "app_{}.R".format(uuid.uuid4().hex))
-                )
+            self._tmp_app_path = os.path.join(
+                "/tmp" if not self.is_windows else os.getenv("TEMP"),
+                uuid.uuid4().hex,
```

Comment:
as @rpkyle  mentioned, the hot reloading on R side will actively monitor all the folder under the main app file, in this case, the old approach using /tmp/app_{}.R is not safe